command/basecommand.php

```diff
@@ -43,34 +49,69 @@ protected function configure() {
 				InputOption::VALUE_NONE,
 				'target all known users'
 			)
+			->addOption(
+				'group',
+				null,
+				InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY,
+				'specify a targeted group to include all users of that group'
+			)
 		;
 		$this->doConfigure();
 	}
 
 	protected function execute(InputInterface $input, OutputInterface $output) {
-		$argsValid = true;
+		$users = array();
 		if (!$input->getOption('all')) {
-			$users = $input->getArgument('user_id');
-
-			if (count($users) === 0) {
-				$output->writeln("Specify either the target user(s) or --all");
-				$argsValid = false;
+			if (count($input->getArgument('user_id'))===0
+					&& count($input->getOption('group'))===0) {
+				$output->writeln("Specify either the target user(s), --group or --all");
+				return;
 			}
 			else {
-				foreach ($users as $user) {
-					if (!$this->userManager->userExists($user)) {
-						$output->writeln("User <error>$user</error> does not exist!");
-						$argsValid = false;
-					}
+				$users = $input->getArgument('user_id');
+				if (!$this->validateUsers($output, $users))
+					return;
```

Comment:
Please always use braces {} around the conditional blocks.

Reply:
Also, it's usually a good idea to avoid having multiple exit points in a function (unless the function is trivially simple). And four levels of nesting in one function is a bit too much for comfort. In this particular case, the cleanest solution would probably be to use exceptions. The function could be something like this:
```
protected function execute($input, $output) {
    try {
        self::ensureUsersGiven($input);
        $argUsers = $this->getArgumentUsers($input);
        $groupUsers = $this->getGroupArgumentUsers($input);
        $users = array_unique(array_merge($argUsers, $groupUsers));
        $this->doExecute($input, $output, $users);
    }
    catch (InvalidArgumentException $e) {
        $output->writeln($e->getMessage());
    }
}

protected static function ensureUsersGiven($input) {
    if (!$input->hasArgument('user_id')
        && !$input->hasOption('all')
        && !$input->hasOption('group')) {
        throw new InvalidArgumentException("Specify either the target user(s), --group or --all");
    }
}
```
The functions `getArgumentUsers` and `getGroupArgumentUsers` would extract the command line arguments and throw an exception if any of the arguments is invalid.